src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/PageSelectorModel.cs

```diff
@@ -63,21 +63,31 @@ public static void PopulateDefaults(PageRouteModel model, string pageRoute, stri
                 return false;
             }
 
-            // Normalize the pages root directory so that it has a 
-            var normalizedPagesRootDirectory = razorPagesOptions.RootDirectory.TrimStart('/');
-            if (!normalizedPagesRootDirectory.EndsWith("/", StringComparison.Ordinal))
+            string normalizedPagesRootDirectory;
+            if (razorPagesOptions.RootDirectory == "/")
```

Comment:
Had to special case empty root "/". I'm going to try and see if I can make a change to StringTokenizer (need the enumerator to expose the string index) and clean this up.